content/en/blog/2024/otel-collector-container-log-parser/index.md

```diff
@@ -0,0 +1,197 @@
+---
+title: Introducing the new container log parser for OpenTelemetry Collector
+linkTitle: Collector container log parser
+date: 2024-05-16
+author: '[Christos Markou](https://github.com/ChrsMark) (Elastic)'
+cSpell:ignore: Christos containerd Filelog filelog Jaglowski kube Markou
+---
+
+[Filelog receiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/filelogreceiver)
+is one of the most commonly used components of the
+[OpenTelemetry Collector](/docs/collector), as indicated by the most recent
+[survey](/blog/2024/otel-collector-survey/#otel-components-usage). According to
+the same survey, it's unsurprising that
+[Kubernetes is the leading platform for Collector deployment (80.6%)](/blog/2024/otel-collector-survey/#deployment-scale-and-environment).
+Based on these two facts, we can realize the importance of seamless log
+collection on Kubernetes environments.
+
+Currently, the
+[filelog receiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/v0.100.0/receiver/filelogreceiver/README.md)
+is capable of parsing container logs from Kubernetes Pods, but it requires
+[extensive configuration](https://github.com/open-telemetry/opentelemetry-helm-charts/blob/aaa70bde1bf8bf15fc411282468ac6d2d07f772d/charts/opentelemetry-collector/templates/_config.tpl#L206-L282)
+to properly parse logs according to various container runtime formats. The
+reason, is that container logs can come in various known formats (according to
+the container runtime), and hence in order to properly parse them we need to
+perform a specific set of operations:
+
+1. detect the format of the incoming logs at runtime
+2. parse each format accordingly taking into account its format specific
+   characteristics: define if it's JSON or plain text and take into the
+   timestamp format
+3. extract known metadata relying on predefined patterns.
+
+Such advanced sequence of operations can be handled by chaining the proper
+[stanza](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/pkg/stanza)
+operators together. The end result is rather complex, and this configuration
+complexity can be mitigated by using the corresponding
+[helm chart preset](https://github.com/open-telemetry/opentelemetry-helm-charts/tree/main/charts/opentelemetry-collector#configuration-for-kubernetes-container-logs).
+However, despite having the preset, it can still be challenging for users to
+maintain and troubleshoot such advanced configurations.
```

Comment:
```suggestion
operators together. The end result is rather complex. This configuration
complexity can be mitigated by using the corresponding
[helm chart preset](https://github.com/open-telemetry/opentelemetry-helm-charts/tree/main/charts/opentelemetry-collector#configuration-for-kubernetes-container-logs).
However, despite having the preset, it can still be challenging for users to
maintain and troubleshoot such advanced configurations.
```

Style nit: In some parts you use first person plural ("we"), in others you switch to passive voice ("it can be..."). I recommend to select a single voice and stick to it, preferably second person, which is more engaging ("You can...").